addons/api/addon/models/credential-library.js

```diff
@@ -20,14 +28,33 @@ export default class CredentialLibraryModel extends GeneratedCredentialLibraryMo
    * @type {boolean}
    */
   get isUnknown() {
-    return !types.includes(this.type);
+    return !TYPES_CREDENTIAL_LIBRARY.includes(this.type);
   }
 
   /**
    * True if credential is a vault type.
    * @type {boolean}
    */
   get isVault() {
-    return this.type === 'vault';
+    return (
+      this.type === TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC ||
+      this.type === TYPE_CREDENTIAL_LIBRARY_VAULT_SSH_CERTIFICATE
+    );
+  }
+
+  /**
+   * True if credential is a generic vault type.
+   * @type {boolean}
+   */
+  get isVaultGeneric() {
+    return this.type === TYPE_CREDENTIAL_LIBRARY_VAULT_GENERIC;
+  }
+
+  /**
+   * True if credential is a vault ssh cert type.
+   * @type {boolean}
+   */
+  get isVaultSSHCert() {
```

Comment:
Minor nit, should this be Certificate fully qualified to reflect the type name elsewhere?

Reply:
Yup missed that